Code/Engine/RendererCore/Textures/TextureLoader.cpp

```diff
@@ -101,43 +102,105 @@ ezResourceLoadData ezTextureResourceLoader::OpenDataStream(const ezResource* pRe
     if (File.Open(pResource->GetResourceID()).Failed())
       return res;
 
-    const ezStringBuilder sAbsolutePath = File.GetFilePathAbsolute();
-    res.m_sResourceDescription = File.GetFilePathRelative().GetView();
+    if (File.GetFilePathAbsolute().HasExtension("ezBinColorGradient"))
+    {
+      res.m_sResourceDescription = File.GetFilePathRelative().GetView();
 
 #if EZ_ENABLED(EZ_SUPPORTS_FILE_STATS)
-    {
-      ezFileStats stat;
-      if (ezFileSystem::GetFileStats(pResource->GetResourceID(), stat).Succeeded())
       {
-        res.m_LoadedFileModificationDate = stat.m_LastModificationTime;
+        ezFileStats stat;
+        if (ezFileSystem::GetFileStats(pResource->GetResourceID(), stat).Succeeded())
+        {
+          res.m_LoadedFileModificationDate = stat.m_LastModificationTime;
+        }
       }
-    }
 #endif
 
-    /// In case this is not a proper asset (ezTextureXX format), this is a hack to get the SRGB information for the texture
-    const ezStringBuilder sName = ezPathUtils::GetFileName(sAbsolutePath);
-    pData->m_TexFormat.m_bSRGB = (sName.EndsWith_NoCase("_D") || sName.EndsWith_NoCase("_SRGB") || sName.EndsWith_NoCase("_diff"));
-
-    if (sAbsolutePath.HasExtension("ezBinTexture2D") || sAbsolutePath.HasExtension("ezBinTexture3D") || sAbsolutePath.HasExtension("ezBinTextureCube") || sAbsolutePath.HasExtension("ezBinRenderTarget") || sAbsolutePath.HasExtension("ezBinLUT"))
-    {
-      if (LoadTexFile(File, *pData).Failed())
+      // skip the asset file header at the start of the file
+      ezAssetFileHeader AssetHash;
+      if (AssetHash.Read(File).Failed())
         return res;
+
+      ezColorGradientResourceDescriptor desc;
+      desc.Load(File);
+
+      pData->m_TexFormat.m_bSRGB = true;
+
+      const ezUInt32 uiHeight = 1;
+
+      // set up a 1024 * 4 uncompressed RGBA texture
+      ezImageHeader header;
+      header.SetWidth(1024);
+      header.SetHeight(uiHeight);
+      header.SetDepth(1);
+      header.SetImageFormat(ezImageFormat::R8G8B8A8_UNORM_SRGB);
```

Comment:
What about HDR gradients?

Reply:
I wasn't sure about those, so I left that problem up to future me.